File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        
        
        dp = [0]*(10**5+1)
        dp[0] = 0 
        for i in range(1,10**5+1):
            dp[i] = i + dp[i-1]
            
        combo =0 
        ans = 0 
        for n in nums: 
            
            if n ==0: 
                combo +=1 
            else: 
                ans += dp[combo]
                combo =0 
                
        if combo:
            ans += dp[combo]
        
        return ans 
            

File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.java
class Solution {
    public long zeroFilledSubarray(int[] nums) {
        
        long ans =0 ; 
        long streak = 0 ; 
        for(var n:nums){
            
            
            if(n==0){
                streak++ ; 
            } else {
                streak = 0 ; 
            }
            ans += streak ; 


        } return ans ;  


    }
}